resources/assets/lib/profile-page/beatmapsets.tsx

```diff
@@ -28,49 +28,58 @@ type Props = {
 export default class Beatmapsets extends React.PureComponent<Props> {
   render() {
     return (
-      <div className='page-extra'>
+      <div className="page-extra">
         <ExtraHeader name={this.props.name} withEdit={this.props.withEdit} />
-        {sectionKeys.map(([section, key]) => this.renderBeatmapsets(section, key))}
+        {sectionKeys.map(([section, key]) =>
+          this.renderBeatmapsets(section, key)
+        )}
       </div>
     );
   }
 
-  private readonly renderBeatmapsets = (section: BeatmapsetSection, key: string) => {
+  private readonly renderBeatmapsets = (
+    section: BeatmapsetSection,
+    key: string
+  ) => {
     const count = this.props.counts[section];
     const beatmapsets = this.props[section];
     const pagination = this.props.pagination[section];
 
     return (
       <React.Fragment key={section}>
-        <h3 className='title title--page-extra-small'>
+        <h3 className="title title--page-extra-small">
           {osu.trans(`users.show.extra.beatmaps.${key}.title`)}
-          <span className='title__count'>{osu.formatNumber(count)}</span>
+          <span className="title__count">{osu.formatNumber(count)}</span>
         </h3>
 
         {beatmapsets.length > 0 && (
-          <div className='osu-layout__col-container osu-layout__col-container--with-gutter js-audio--group'>
+          <div className="osu-layout__col-container osu-layout__col-container--with-gutter js-audio--group">
             {beatmapsets.map((beatmapset) => (
               <div
                 key={beatmapset.id}
-                className='osu-layout__col osu-layout__col--sm-6'
+                className="osu-layout__col osu-layout__col--sm-6"
               >
-                <BeatmapsetPanel beatmapset={observable(beatmapset)} />
+                <BeatmapsetPanel
+                  beatmapset={observable(beatmapset)}
+                  currentUserName={this.props.user.username}
+                  section={section}
```

Comment:
the logic of deciding to display guest badge should be local to the profile page, so figure it out here instead of in `BeatmapsetPanel` and then only pass a flag into `BeatmapsetPanel`